roles/chameleon_mariadb/templates/mariadb.j2

```diff
@@ -32,21 +36,34 @@ ADDITIONAL_OPTIONS="--ignore-table=mysql.event"
 
 ##### STOP CONFIG ####################################################
 
-set -o pipefail
-
 backup_file=${DIR}/${PREFIX}`date +%Y%m%d-%H%M%S`.sql.bz2
 
 cleanup() {
   rm -f "$TMPFILE"
   # Rotate backups
   find "${DIR}/" -maxdepth 1 -type f -name "${PREFIX}*.sql*" -mtime +${ROTATE} -print0 | xargs -0 -r rm -f
 }
-trap cleanup EXIT
 
 mkdir -p $DIR
 mysqldump --defaults-extra-file="$TMPFILE" --opt --flush-logs --single-transaction \
  ${ADDITIONAL_OPTIONS} \
  --all-databases | bzcat -zc > "$backup_file"
+{% endif %}
+
+{% if enable_mariabackup %}
+
+DIR=$(docker volume inspect mariadb_backup | jq -r '.[].Mountpoint')
+PREFIX=mysqlbackup-
+backup_file="${DIR}/$(ls -t1 ${DIR}/${PREFIX}* |  head -n 1)"
+
+cleanup() {
+  # Rotate backups
+  find "${DIR}/" -maxdepth 1 -type f -name "${PREFIX}*" -mtime +${ROTATE} -print0 | xargs -0 -r rm -f
```

Comment:
This may not work very well if the volume is not bind-mounted to the host (e.g., it could be provided by NFS or RBD or something). It may be better to run this through Docker:

```shell
docker run --rm -v mariadb_backup:/backup {{ kolla_toolbox_image_full }} \
  find /backup -maxdepth 1 -type f -name "${PREFIX}*" -mtime +${ROTATE} -print0 \
  | xargs -0 -r -rm -f
```

The above creates a new instance of a toolbox container (could be any base container, maybe even the mariadb container image, which tehre should be a variable for in scope) and mounts the backup at /backup, then deletes it inside the container.